src/java/org/apache/cassandra/index/sai/metrics/TableQueryMetrics.java

```diff
@@ -59,6 +67,12 @@ public void record(QueryContext queryContext)
             totalQueryTimeouts.inc();
```

Comment:
nit: Quick message on the assert above?

Reply:
This is a bit dumb really. We are only expecting the `queryTimeouts` to be `0` or `1` because if it is incremented we throw an exception that kills the query.

I've simplified this by changing `long queryTimeouts` to `boolean queryTimedOut` and removed the assert.